### PR TITLE
Ember Fastboot rc.1 support

### PR DESCRIPTION
**Pull request by mazondo**
Ember fastboot rc.1 has some breaking changes that need to be handled.  More info here:

https://gist.github.com/kratiahuja/d22de0fb1660cf0ef58f07a6bcbf1a1c

### CODE REVIEW

**Comment by mazondo** [comment]
forgot to mention I also moved swiper to package.json dep

**Comment by mschinis** [comment]
@Suven can we merge this one in?

**Comment by Suven** [comment]
Hey, 

I wanted to test how this affects ppl who already have the bower-dep installed when updating, but didn't manage to do so.

I'll try my best to merge this by tomorrow and release a new version on npm afterwards

**Comment by mazondo** [comment]
Probably should have submitted this as 2 prs, sorry about that. Got excited cause this was our last bower dep 😂

File: index.js [hunk 1]
```diff
@@ -1,32 +1,35 @@
...
-      } while (current.parent.parent && (current = current.parent));
-    }
+    browserVendorLib = map(browserVendorLib, (content, relativePath) => {
+      if (relativePath.indexOf('css') !== -1) {
+        return content;
+      }
+      return `if (typeof FastBoot === 'undefined') { ${content} }`
+    });
+  
+    return new mergeTrees([defaultTree, browserVendorLib]);
```
Comment (jelhan) [inline index.js]: Faced an issue recently that `defaultTree` was `undefined` and therefore `mergeTrees` throws. `return browserVendorLib;` might be fine here.

Comment (Matt-Jensen) [inline index.js]: I think following ember-gestures lead, merging the vendor tree if it is present, would work fine:

```js
treeForVendor: function(defaultTree) {
  var trees = [];
  var swiperPath = path.join(path.dirname(require.resolve('swiper')), '..');
  var browserVendorLib = new Funnel(swiperPath, {
    destDir: 'swiper',
    include: ['js/swiper.min.js', 'css/swiper.min.css']
  });

  browserVendorLib = map(browserVendorLib, (content, relativePath) => {
    if (relativePath.indexOf('css') !== -1) {
      return content;
    }
    return `if (typeof FastBoot === 'undefined') { ${content} }`
  });

  if (defaultTree !== undefined) {
    trees.push(defaultTree);
  }

  return new mergeTrees(trees);
},
```

This is the also the approach I used to be Fastboot v1 compatible in my addon ember-cli-g-maps.

Comment (Matt-Jensen) [inline index.js]: Btw this same code is successfully installing the npm dependency on my fork: https://github.com/Matt-Jensen/ember-cli-swiper/tree/latest

**Comment by Matt-Jensen** [comment]
Noticed this PR still has a blueprint which adds the original swiper bower dependency after install.  @outdoorsy running: `ember destroy blueprint ember-cli-swiper` and then committing to your fork should do the trick.

**Comment by topaxi** [comment]
Any news on this one? I'd be happy to help out if something is blocking this PR.

**Comment by mazondo** [comment]
Sorry, missed this note.  I've just made the change recommended by @Matt-Jensen .  TIL, thanks for that!  Hopefully that's all that's missing here, I'd love to get back on master.

**Comment by Suven** [comment]
I'll merge a few more things tomorrow, do a retest with all new PR's and then ping this PR when the new release is done in npm.

Thanks to everybodys contribution!

**Comment by Matt-Jensen** [comment]
@Suven I see a potential issue with the merged commit [d426a4f](https://github.com/Suven/ember-cli-swiper/commit/d426a4f1e1a7ca7de2742d203f8b615198822cc9) which I believe relates to #62 . We need to be checking if the `defautTree` is defined before merging it into the vendor tree.  Please see [my fork for an example](https://github.com/Matt-Jensen/ember-cli-swiper/blob/latest/index.js).

**Pull request merged**